pyteal/types.py

```diff
@@ -39,15 +39,15 @@ def valid_base32(s:str):
     """
     pattern = re.compile(r'[A-Z2-9]*') # RFC 4648 base 32 w/o padding
 
-    if pattern.fullmatch(s) is None:
+    if len(s) == 0 or pattern.fullmatch(s) is None:
         raise TealInputError("{} is not a valid RFC 4648 base 32 string".format(s))
 
 def valid_base64(s:str):
     """ check if s is a valid base64 encoding string
     """
     pattern = re.compile(r'^(?:[A-Za-z0-9+/]{4})*(?:[A-Za-z0-9+/]{2}==|[A-Za-z0-9+/]{3}=)?$')
 
-    if pattern.fullmatch(s) is None:
+    if len(s) == 0 or pattern.fullmatch(s) is None:
```

Comment:
I believe that the TEAL assembler accepts empty strings.

Also, to be pedantic about the error message, see here: https://stackoverflow.com/questions/15212127/is-an-empty-string-valid-base64-encoded-data-of-zero-bytes-length

Reply:
You are correct. It looks like `byte base64` is not allowed, but `byte base64()` does create an empty string. I will remove this restriction.

Reply:
It might also make sense to alter `byte base32` and just make the TEAL assembler consistent in all encodings.